js/article.js

```diff
@@ -1,49 +1,117 @@
-var articles = [];
+// CREATING PULL REQUEST FOR SELF-REVIEW/ CLASS_9 PORTFOLIO ASSIGNMENT
+/* TODO:
+OVERALL:
+1) MAKE SURE FOLDERS/FILES ARE ORGANIZED LOGICALLY.
+2) MAKE SURE ALL FILES ARE LINKED PROPERLY AND LISTED IN RIGHT ORDER.
+ARTICLE.JS:
+1) I NEED TO SEE IF THERE ARE OTHER WAYS TO USE HANDLEBARS  AND IF I AM EFFICIENTLY USING THIS AS POSSIBLE TO MAKE MY CODE EFFICIENTLY
+2) I NEED TO UPDATE .PUBLISHEDON TO APPLY MORE TO MY PORTFOLIO--> IE. DISPLAY COMPLETION DATE FOR EACH PROJECT IN DECENDING ORDER
+3) ADD MORE COMMENTS SO I CAN LOGICALLY FOLLOW/APPRECIATE MY CODE AND ALL FOLDERS/FILES, SO I CAN TRULY APPRECIATE HOW THEY ARE DYNAMICALLY WORKING TOGETHER
+4) GET RID OF ANY GHOST CODE, REFACTOR WHERE POSSIBLE*/
 
-function Article (opts) {
-  // TODO: DONE- Use the js object passed in to complete this constructor function:
-  // Save all the properties of 'opts' into 'this'
-  this.author = opts.author;
-  this.title = opts.title;
-  this.category = opts.category;
-  this.publishedOn = opts.publishedOn;
-  this.body = opts.body;
-  this.authorUrl = opts.authorUrl;
-}
+// WRAP ENTIRE CONTENTS OF THIS FILE IN AN IIFE. PASS IN MODULE TO IFFE (which objects can be attached for later access)
+(function(module) {
+  function Article (opts) {
+    for (key in opts) {
+      this[key] = opts[key];
+    }
+  }
 
-Article.allArticles = [];
+//ALL ARTICLES ARRAY
+  Article.allArticles = [];
 
-// ADD PROPERTIES USED BY TEMPLATE . EXCUTE LOGIC HERE SINCE TEMPLATE CAN'T HOLD JS LOGIC. ADD RESULT TO OBJECT AS NEW PROPERTY BY KEY IN TEMPLATE. //
-Article.prototype.toHtml = function(scriptTemplateId) {
-  var templateRender = Handlebars.compile($(scriptTemplateId).html());
+// ADD PROPERTIES USED BY TEMPLATE. EXCUTE LOGIC HERE SINCE TEMPLATE CAN'T HOLD JS LOGIC. ADD RESULT TO OBJECT AS NEW PROPERTY BY KEY IN TEMPLATE.
+  Article.prototype.toHtml = function(scriptTemplateId) {
+    var template = Handlebars.compile($(scriptTemplateId).text());
+    var minuteRead = Math.floor(Article.getNumWords(this) / 200);   // calculate article read time using average reader speed (200wpm)
+    this.readLength = (minuteRead <= 1) ? '1 minute' : minuteRead + ' minutes';
```

Comment:
The minutes it would take to read is a very cool feature. I think that this a nice addition and an interesting piece of code.